### PR TITLE
Introducing incremental GitHubRepository Entity Provider

### PR DESCRIPTION
**Pull request by taras**
## Motivation

By providing documentation, we want to make it easier for people to write Incremental Entity Providers. 

## Approach

Did a lot of polishing to do writing, using, and installing Incremental Entity Providers easy.

giphy-12

### CODE REVIEW

**Comment by frontsidejack** [comment]
The following preview packages were published:
|                                         Name                                         |                                        Version                                       | Install Command                                                                                                                                |
| :----------------------------------------------------------------------------------: | :----------------------------------------------------------------------------------: | ---------------------------------------------------------------------------------------------------------------------------------------------- |
| <br><pre><code>@frontside/backstage-plugin-incremental-ingestion-github</code></pre> | <br><pre><code>0.1.1-tm-incremental-github-repository-entity-provider.7</code></pre> | <br><pre><code>yarn add @frontside/backstage-plugin-incremental-ingestion-github@tm-incremental-github-repository-entity-provider</code></pre> |
<p align='right'><em>Generated by <a href='https://github.com/thefrontside/actions/tree/main/publish-pr-preview'>@thefrontside/actions Frontside</a><em></p>

File: plugins/incremental-ingestion-backend/README.md [hunk 1]
```diff
@@ -32,6 +32,285 @@ This approach has the following benefits,
...
+export class MyIncrementalEntityProvider implements IncrementalEntityProvider<Cursor, Context> {
+  getProviderName() {
+    return `MyIncrementalEntityProvider`;
+  }
+
+  async around(burst: (context: MyContext) => Promise<void>): Promise<void> {
+
+    const apiClient = new MyApiClient();
+
+    await around({ apiClient });
```
Comment (cowboyd) [inline plugins/incremental-ingestion-backend/README.md]:
```suggestion
    await burst({ apiClient });
```

File: plugins/incremental-ingestion-backend/README.md [hunk 2]
```diff
@@ -32,6 +32,285 @@ This approach has the following benefits,
...
+  getProviderName() {
+    return `MyIncrementalEntityProvider`;
+  }
+
+
+  async around(burst: (context: MyContext) => Promise<void>): Promise<void> {
+
+    const apiClient = new MyApiClient(this.token)
+
+    await around({ apiClient })
```
Comment (cowboyd) [inline plugins/incremental-ingestion-backend/README.md]:
```suggestion
    await burst({ apiClient })
```

File: plugins/incremental-ingestion-backend/README.md [hunk 3]
```diff
@@ -32,6 +32,285 @@ This approach has the following benefits,
...
```
Comment (cowboyd) [inline plugins/incremental-ingestion-backend/README.md]:
```suggestion
    await burst({ apiClient })
```

**Review by cowboyd — CHANGES_REQUESTED** [review]
Looks amazing! Have a few fixes to the docs

**Review by cowboyd — APPROVED** [review]
Absolutely bang-up work..

**Pull request merged**